SECURITY.md

```diff
@@ -0,0 +1,59 @@
+
+1. Report it on Github directly you can follow the procedure described
+[here](https://docs.github.com/en/code-security/security-advisories/guidance-on-reporting-and-writing/privately-reporting-a-security-vulnerability)
+and:
+    - Navigate to the security tab on the repository
+    - Click on 'Advisories'
+    - Click on 'Report a vulnerability'
+
+2. Send an email to `security@crossplane.io` detailing the issue and steps
+to reproduce.
+
+The reporter(s) can expect a response within 24 hours acknowledging
+the issue was received. If a response is not received within 24 hours, please
+reach out to any committer directly to confirm receipt of the issue.
+
+## Review Process
+
+Once a committer has confirmed the relevance of the report, a draft security
+advisory will be created on Github. The draft advisory will be used to discuss
+the issue with committers, the reporter(s), and crossplane's security advisors.
+If the reporter(s) wishes to participate in this discussion, then provide
+reporter Github username(s) to be invited to the discussion. If the reporter(s)
+does not wish to participate directly in the discussion, then the reporter(s)
+can request to be updated regularly via email.
+
+If the vulnerability is accepted, a timeline for developing a patch, public
+disclosure, and patch release will be determined. If there is an embargo period
+on public disclosure before the patch release, an announcment will be sent to
+the security announce mailing list (`crossplane-security-announce@lists.cncf.io`)
+announcing the scope of the vulnerability, the date of availability of the
+patch release, and the date of public disclosure. The reporter(s) are expected
+to participate in the discussion of the timeline and abide by agreed upon dates
+for public disclosure.
+
+## Supported Versions
+
+See the [crossplane documentation](https://docs.crossplane.io/master/reference/release-cycle/)
+for information on supported versions of crossplane. Any supported
+release branch may receive security updates. For any security issues discovered
+on older versions, non-core packages, or dependencies, please inform committers
+using the same security mailing list as for reporting vulnerabilities.
+
+## Joining the security announce mailing list
+
+Any organization or individual who directly uses crossplane and non-core
+packages in production or in a security critical application is eligible to join
+the security announce mailing list. Indirect users who use crossplane through a
+vendor are not expected to join, but should request their vendor join. To join
+the mailing list, the individual or organization must be sponsored by either a
+crossplane committer or security advisor as well as have a record of properly
+handling non-public security information. If a sponsor cannot be found,
+sponsorship may be requested at `security@crossplane.io`. Sponsorship should not
+be requested via public channels since membership of the security announce list
+is not public.
```

Comment:
Same as above, might be too much at the moment

Reply:
I'm fine with anyone requesting to join the announcement list - I don't see a need at this point to restrict it heavily since it will be used to broadcast vulnerabilities along with their fixes, to get people to update to the fix.

Maybe we can shorten this to:

```
Any organization or individual wishing to join the security announce mailing list, please send a request to `security@crossplane.io`.
```

Reply:
If I got it right, the point here is not to have vulnerabilities disclosed alongside the announcement of the fixed releases, but to have a mailing list from which subscribed users could get notified with vague descriptions of vulnerabilities not yet public, so that they can be aware something is coming. On one side it should not be considered secret as it is not and secrecy is not a valid security strategy, but it should not be too much advertised either.

Reply:
ah OK @phisco, I think I didn't catch that intention of the announcement list. To make it's intent more clear, perhaps we can add some more language to the beginning of this paragraph like:

> The security announcement list will be used to announce vulnerabilities, often ahead of when a fix is made available, to a restricted set of Crossplane adopters and vendors.

And with that, I'm fine with the restrictions for joining as currently proposed :)

Reply:
I see now that there is already some language regarding this announcement list in the "review process" section, so you can choose whether you want to add this extra language in the "Joining the security announce mailing list" section or not 🤪

Reply:
added, better to be explicit 😉